scripts/convert-ESO-PV-forecasts-from-CSV-to-HDF5.py

```diff
@@ -0,0 +1,195 @@
+
+    return eso_forecasts_df
+
+
+def keep_only_2_day_forecasts(eso_forecasts_df: pd.DataFrame) -> pd.DataFrame:
+    """Throw away any rows where FORECAST_HORIZON != '2D'.
+
+    ESO run two forecasts:  2D (2-day ahead) and 14D (14-day ahead).
+    We only want the 2D forecasts because we're comparing against OCF's nowcasts.
+    """
+    eso_forecasts_df.FORECAST_HORIZON = eso_forecasts_df.FORECAST_HORIZON.str.upper()
+    rows_2D_horizon = eso_forecasts_df.FORECAST_HORIZON == "2D"
+    eso_forecasts_df = eso_forecasts_df[rows_2D_horizon]
+    eso_forecasts_df.drop(columns='FORECAST_HORIZON', inplace=True)
+    return eso_forecasts_df
+
+
+def split_asl_and_ml_forecasts(eso_forecasts_df: pd.DataFrame) -> dict[str, pd.DataFrame]:
+    """Split the forecasts into ASL forecasts, and ML forecasts.
+
+    Returns a dict where the keys are the ESO forecasting algorithm name 'ASL' or 'ML',
+    and the values are the pd.DataFrame of forecast data.
+    """
+    eso_forecasts_df.SCRIPT_NAME = eso_forecasts_df.SCRIPT_NAME.str.upper()
+
+    eso_forecasts = {}
+    for algo_name in ESO_ALGO_NAMES:
+        mask = eso_forecasts_df.SCRIPT_NAME.str.contains(f'_{algo_name}_')
+        eso_forecasts[algo_name] = eso_forecasts_df[mask]
+        eso_forecasts[algo_name] = eso_forecasts[algo_name].drop(columns='SCRIPT_NAME')
+
+    return eso_forecasts
+
+
+def convert_to_dataarray(df: pd.DataFrame) -> xr.DataArray:
+    """Convert forecast data to a 3-dimensional DataArray, with these dimensions:
+
+    1. `gsp_id`: A string identifying the Grid Supply Point region.
+    2. `forecast_date_time`: The UTC datetime when ESO ran their forecast.  Not exactly the
+        same as the NWP init time.  This script takes the floor('30T') of the
+        original forecast_date_time from ESO.
+    3. `step`: The Timedelta between the forecast_date_time and the target_date_time.
+    """
+    df = df.copy()  # So we don't modify the passed-in `df` object.
+
+    # The FORECAST_DATE_TIME is, I think, the time when ESO ran their forecasting script.
+    # The times are all over the place.  So floor the datetime to the nearest half-hour:
+    df['forecast_date_time'] = df['FORECAST_DATE_TIME'].dt.floor('30T')
+    df = df.drop(columns='FORECAST_DATE_TIME')
+
+    # Calculate the forecast "step":
+    df['step'] = df['TARGET_DATE_TIME'] - df['forecast_date_time']
+    df = df.drop(columns='TARGET_DATE_TIME')
+
+    # Make sure "step" is positive:
+    # (step can be negative for ASL forecasts, for some reasons).
+    df = df[df.step >= pd.Timedelta(0)]
+
+    # Rename to more column names more like the ones we're used to.
+    df = df.rename(columns={'SITE_ID': 'gsp_id'})
+
+    # Set index
+    df = df.set_index(['gsp_id', 'forecast_date_time', 'step'])
+    series = df.squeeze()
+    del df
+
+    # Remove duplicate indicies
+    duplicated = series.index.duplicated()
+    series = series[~duplicated]
```

Comment:
could you reduce it to one line using - https://pandas.pydata.org/docs/reference/api/pandas.Series.drop_duplicates.html